tools/run_tests/xds_k8s_test_driver/framework/infrastructure/k8s_internal/k8s_log_collector.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2022 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+import os
+import pathlib
+import threading
+from typing import Any, Callable, Optional, TextIO
+
+from kubernetes import client
+from kubernetes.watch import watch
+
+logger = logging.getLogger(__name__)
+
+
+class PodLogCollector(threading.Thread):
```

Comment:
There are [a bunch of landmines](https://stackoverflow.com/questions/660961/overriding-python-threading-thread-run) when subclassing `threading.Thread` rather than just passing a callable to the `threading.Thread` constructor, but it looks like you've managed to avoid them all. No complaints on my end as long you've confirmed that it's all working as intended.

Reply:
Yea, I had an impression this was the case, but couldn't find any strong advices against it. Thanks for noticing! I'll give it another looks, just in case. Won't be a difficult refactoring to not extend a thread, but manage it as a field.